packages/astro/src/actions/index.ts

```diff
@@ -50,6 +60,42 @@ export default function astroActions({ fs = fsMod }: { fs?: typeof fsMod }): Ast
 	};
 }
 
+/**
+ * This plugin is responsible to load the known file `actions/index.js` / `actions.js`
+ * If the file doesn't exist, it returns an empty object.
+ * @param settings
+ */
+export function vitePluginUserActions({ settings }: { settings: AstroSettings }): VitePlugin {
+	let resolvedActionsId: string;
+	return {
+		name: '@astro/plugin-actions',
+		async resolveId(id) {
+			if (id === NOOP_ACTIONS) {
+				return NOOP_ACTIONS;
+			}
+			if (id === VIRTUAL_INTERNAL_MODULE_ID) {
+				const resolvedModule = await this.resolve(
+					`${decodeURI(settings.config.srcDir.pathname)}actions`
```

Comment:
```suggestion
decodeURI(new URL('actions', settings.config.srcDir).pathname)
```

Nit: could use a URL constructor instead of string concat? Not certain if this would work without testing